experimental/go-dep/image/Dockerfile-stack

```diff
@@ -0,0 +1,35 @@
+FROM golang:1
+
+ENV APPSODY_MOUNTS=/:/project/user-app
```

Comment:
This could be `/:/go/src/github.com/example/user-app` to fix the symlinking... but even then, the location in GOPATH is supposed to depend on the module name. Is there a way to force the user to provide a module name to `appsody init`? The more I stare at this, the more I ask myself - Is there any actual value in allowing GOMODULE aware and unaware stacks? Generally, they correspond to different local developer habits, but in this case, we aren't allowing people to develop locally (except for the "edit src" part of develop), they are being forced to run in a container. Local non-gomodule aware development would involve manual fetch and install of deps with "go get", but that isn't going to work so well with an ephemeral docker dev-time container, their deps will all be lost on container restart. They pretty much have to use go.mod so that the in-container appsody runtime can know what deps to pull in, don't they? Or else, vendor everything, in which case, why not stay out of GOPATH and use `go mod vendor`? I think the README needs updating to explain how go dependencies are intended to be handled with this stack.